src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -165,12 +184,12 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
         return winrt::to_hstring(time_str);
     }
 
-    void ExtensionPalette::_splitResponseAndAddToChatHelper(const winrt::hstring& response, const ErrorTypes errorType)
+    void ExtensionPalette::_splitResponseAndAddToChatHelper(const IResponse response)
     {
         // this function is dependent on the AI response separating code blocks with
         // newlines and "```". OpenAI seems to naturally conform to this, though
         // we could probably engineer the prompt to specify this if we need to.
-        std::wstringstream ss(response.c_str());
+        std::wstringstream ss(response.Message().c_str());
```

Comment:
This function isn't particularly ideal overall, due to C-string usage (this line), stringstream & getline usage (should be a split iterator), as well as calling `find()` up to twice per iteration. The latter should also be `starts_with` instead, which is the actual intention of the code as it checks for `== 0` right after. Should be cleaned up at some point, ideally, but perhaps it goes away anyway with the markdown work.

Reply:
Yeah this whole section is going to be overhauled with Mike's markdown work, so just going to leave it as it is for now